# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xmule/gcc41.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur xmule-1.12.2-orig/src/ED2KLink.h xmule-1.12.2/src/ED2KLink.h
--- xmule-1.12.2-orig/src/ED2KLink.h	2005-12-02 23:05:57.000000000 +0000
+++ xmule-1.12.2/src/ED2KLink.h	2006-03-30 18:05:29.962739500 +0000
@@ -68,7 +68,7 @@
 class CED2KFileLink : public CED2KLink
 {
 public:
-    CED2KFileLink::CED2KFileLink(const TCHAR* name,const TCHAR* size, const TCHAR* hash,const TCHAR* sources);
+    CED2KFileLink(const TCHAR* name,const TCHAR* size, const TCHAR* hash,const TCHAR* sources);
     virtual ~CED2KFileLink();
 
     virtual LinkType GetKind() const;
diff -Nur xmule-1.12.2-orig/src/PPgNotify.h xmule-1.12.2/src/PPgNotify.h
--- xmule-1.12.2-orig/src/PPgNotify.h	2005-12-02 23:05:57.000000000 +0000
+++ xmule-1.12.2/src/PPgNotify.h	2006-03-30 18:31:56.377884250 +0000
@@ -49,7 +49,7 @@
 	CPreferences* app_prefs;
 protected:
 	//virtual void DoDataExchange(CDataExchange* pDX);    // Supporto DDX/DDV
-	wxString CPPgNotify::DialogBrowseFile(wxString Filters, wxString DefaultFileName="");
+	wxString DialogBrowseFile(wxString Filters, wxString DefaultFileName="");
 
 	//DECLARE_MESSAGE_MAP()
 	DECLARE_EVENT_TABLE()
diff -Nur xmule-1.12.2-orig/src/PartFile.h xmule-1.12.2/src/PartFile.h
--- xmule-1.12.2-orig/src/PartFile.h	2005-12-02 23:05:57.000000000 +0000
+++ xmule-1.12.2/src/PartFile.h	2006-03-30 18:00:04.790417500 +0000
@@ -113,8 +113,8 @@
     bool IsText();
 
 
-    wxString CPartFile::getPartfileStatus(); //<<--9/21/02
-    int32_t CPartFile::getTimeRemaining(); //<<--9/21/02
+    wxString getPartfileStatus(); //<<--9/21/02
+    int32_t getTimeRemaining(); //<<--9/21/02
     time_t lastseencomplete;
     int getPartfileStatusRang();
     wxString GetDownloadFileInfo();
diff -Nur xmule-1.12.2-orig/src/SafeFile.h xmule-1.12.2/src/SafeFile.h
--- xmule-1.12.2-orig/src/SafeFile.h	2005-12-02 23:05:57.000000000 +0000
+++ xmule-1.12.2/src/SafeFile.h	2006-03-30 18:00:40.656659000 +0000
@@ -25,14 +25,14 @@
 class CSafeFile : public CFile{
 public:
 	CSafeFile();
-	CSafeFile::CSafeFile(x::LPCSTR lpszFileName,x::UINT nOpenFlags);
+	CSafeFile(x::LPCSTR lpszFileName,x::UINT nOpenFlags);
 	x::UINT Read(void* lpBuf,x::UINT nCount);
 };
 
 class CSafeMemFile : public CMemFile{ // memFile
 public:
 	CSafeMemFile(x::UINT nGrowBytes = 0);
-	CSafeMemFile::CSafeMemFile(x::BYTE* lpBuffer,x::UINT nBufferSize,x::UINT nGrowBytes = 0);
+	CSafeMemFile(x::BYTE* lpBuffer,x::UINT nBufferSize,x::UINT nGrowBytes = 0);
 	x::UINT Read(void* lpBuf,x::UINT nCount);
 };
 #endif
diff -Nur xmule-1.12.2-orig/src/ini2.cpp xmule-1.12.2/src/ini2.cpp
--- xmule-1.12.2-orig/src/ini2.cpp	2005-12-02 23:05:57.000000000 +0000
+++ xmule-1.12.2/src/ini2.cpp	2006-03-30 18:28:46.694029750 +0000
@@ -708,8 +708,10 @@
     while ((loc = buffer.find(",")) != std::string::npos)
     {
         std::istringstream sin;
-        
-        sin.str(buffer.substr(0, loc));
+
+	std::string qiu = buffer.substr(0, loc);        
+	sin.str(qiu);
+//        sin.str(buffer.substr(0, loc));
         sin >> holder;
 
         out_array.push_back(holder);
